hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/FSQueueConverter.java

```diff
@@ -309,6 +311,11 @@ private void checkMaxChildCapacitySetting(FSQueue queue) {
     }
   }
 
+  private boolean checkAutoQueueCreationV2Disabled(String queueName) {
+    return !Objects.equals(capacitySchedulerConfig.get(
+        PREFIX + queueName + DOT + AUTO_QUEUE_CREATION_V2_ENABLED), "true");
```

Comment:
This won't return true because Object.equals only returns true if both of the compared objects refer to the same object, String.equals compares the values themselves. But CapacitySchedulerConfiguration already has a helper method which parses the value to a boolean, so I suggest using that. See isAutoQueueCreationV2Enabled. 

Also if percentage mode is used when converting it'll have the legacy Auto Queue Creation property, which too has a helper method, called isAutoCreateChildQueueEnabled. Both of these should be checked.